replica/table.cc

```diff
@@ -3505,16 +3547,11 @@ future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locato
     auto holder = async_gate().hold();
 
     auto& sg = storage_groups()[tid.value()];
-
+    if (!sg) {
+        throw std::runtime_error(format("Cannot cleanup tablet {} of table {}.{} because it is not allocated in this shard",
+                                        tid, _schema->ks_name(), _schema->cf_name()));
+    }
     for (auto& cg_ptr : sg->compaction_groups()) {
-        if (!cg_ptr) {
-            throw std::runtime_error(format("Cannot cleanup tablet {} of table {}.{} because it is not allocated in this shard",
-                                            tid, _schema->ks_name(), _schema->cf_name()));
-        }
-
-        // Synchronizes with in-flight writes if any, and also takes care of flushing if needed.
-        // FIXME: to be able to stop group and provide guarantee above, we must first be able to reallocate a new group if tablet is migrated back.
-        //co_await _cg.stop();
         co_await cg_ptr->flush();
```

Comment:
why did you have to move stop() to update_effective_replication_map()? I think it belongs here as we want to synchronize cleanup with in-flight writes and compaction.